modules/rpc/handler.go

```diff
@@ -97,3 +98,41 @@ func (f fxContextOnewayInboundMiddleware) HandleOneway(ctx context.Context, req
 	fxctx := fxcontext.New(ctx, f.Host)
 	return handler.HandleOneway(fxctx, req)
 }
+
+type authInboundMiddleware struct {
+	service.Host
+}
+
+func (a authInboundMiddleware) Handle(ctx context.Context, req *transport.Request, resw transport.ResponseWriter, handler transport.UnaryHandler) error {
+	fxctx, err := authorize(ctx, a.Host)
+	if err != nil {
+		return err
```

Comment:
Just chatted with @abhinav - you can actually return a `yarpc.api.transport.BadRequestError` which will get treated as a 4xx style error. This is important because Authz/n doesn't mean the server is unhealthy, it means the client is err'ing.